packages/gatsby/src/schema/schema.js

```diff
@@ -121,25 +123,29 @@ const processTypeComposer = async ({
   nodeStore,
   parentSpan,
 }) => {
-  if (
-    typeComposer instanceof ObjectTypeComposer &&
-    typeComposer.hasInterface(`Node`)
-  ) {
-    await addFieldResolvers({ schemaComposer, typeComposer, parentSpan })
-    await addNodeInterfaceFields({ schemaComposer, typeComposer, parentSpan })
-    await addResolvers({ schemaComposer, typeComposer, parentSpan })
-    await addConvenienceChildrenFields({
-      schemaComposer,
-      typeComposer,
-      nodeStore,
-      parentSpan,
-    })
-    await addTypeToRootQuery({ schemaComposer, typeComposer, parentSpan })
-  } else if (typeComposer instanceof ObjectTypeComposer) {
-    await addFieldResolvers({ schemaComposer, typeComposer, parentSpan })
+  if (typeComposer instanceof ObjectTypeComposer) {
+    await processFieldExtensions({ schemaComposer, typeComposer, parentSpan })
+    if (typeComposer.hasInterface(`Node`)) {
+      await addNodeInterfaceFields({ schemaComposer, typeComposer, parentSpan })
+      await addResolvers({ schemaComposer, typeComposer, parentSpan })
+      await addConvenienceChildrenFields({
+        schemaComposer,
+        typeComposer,
+        nodeStore,
+        parentSpan,
+      })
+      await addTypeToRootQuery({ schemaComposer, typeComposer, parentSpan })
+    }
   }
 }
 
+const registerExtensions = ({ parentSpan }) =>
```

Comment:
No reason to have separate API. Let's make it an action like `createTypes`.

Reply:
Let's first quickly discuss how happy we are with `createTypes` being an action? The issue is that actions get passed to all Node APIs but we need it to be called before schema generation, that's why `createTypes` should really only be called in `sourceNodes` even though it is available in other APIs as well. But I'm happy if we make this the general convention to follow.

Reply:
Okay, let's do it this way:

1. Add deprecation warnings to actions for createTypes  and third party schema adding.
2. Create a schema customization runner API that is called after source nodes. Have createTypes, thirdPartySchemas and this new stuff available there.
3. Add all further schema customization stuff there.
4. Update source-graphql to use 3 instead of 1.

Reply:
Sounds good! Probably should be a separate PR?

Reply:
Separate PR sounds good!

Reply:
Yeah, let's merge the thing without customization, then have a separate PR together with Redux changes.